controllers/front/ProductController.php

```diff
@@ -88,6 +88,20 @@ public function canonicalRedirection(string $canonical_url = ''): void
             $this->id_product_attribute = null;
         }
 
+        // Preserve preview parameters if we're in preview mode or if preview parameters are in the URL
+        $extraParams = [];
+        if ($this->isPreview() || (isset($_GET['preview']) && $_GET['preview'] == '1')) {
```

Comment:
```suggestion
        if ($this->isPreview()) {
```

I think this check is enough by taking a look at:
```
if ('1' === Tools::getValue('preview')) {
            $adminTokenValidator = $this->getContainer()->get(LegacyAdminTokenValidator::class);
            $isAdminTokenValid = $adminTokenValidator->isTokenValid((int) Tools::getValue('id_employee'), Tools::getValue('adtoken'));
            if ($isAdminTokenValid) {
                $this->setPreviewMode();
            }
        }
```
I'm also not sure why are you forced to add this code again, you added it inside `getProductLink` which is called in the canonical redirection, so you have those checks already there 🤔

Reply:
@tleon @kpodemski I think the product canonical URL should not change. If its preview or not, doesnt matter. The URL should be the same.

Reply:
let's wait for @tleon feedback, I tested his change previously and it worked, I don't see a reason to change the canonical URL

Reply:
I was trying something to fix my failing UI test but I will revert it did not help me :/